### PR TITLE
Better lexer symbol

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This code is (partially) ported from VFPT and it's main goal is to find proper range of a symbol at position. The previous implementation was buggy, it returned wrong ranges for operators and type parameters.

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/Common/Pervasive.fs [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 [<RequireQualifiedAccess>]
-[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module Option =
-
     /// Gets the value associated with the option or the supplied default value.
     let inline getOrElse v = function
         | Some x -> x | None -> v
 
+    /// Gets the option if Some x, otherwise try to get another value
+    let inline orTry f =
```
Comment (forki) [inline vsintegration/src/FSharp.Editor/Common/Pervasive.fs]: these are in the FSharp.Core now, right? (just as possible optimization)

Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.Editor/Common/Pervasive.fs]: No they are not. I could not compile without these helpers.

Comment (forki) [inline vsintegration/src/FSharp.Editor/Common/Pervasive.fs]: weird. thanks for checking

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
The options module has some additions, the names are a bit different from the ones you used:

````
I think getOrElse becomes defaultValue :
````
https://github.com/Microsoft/visualfsharp/blob/master/src/fsharp/FSharp.Core/option.fs#L19

````
I think orTry looks like orElseWith
````
https://github.com/Microsoft/visualfsharp/blob/master/src/fsharp/FSharp.Core/option.fs#L28

Hope this helps

Kevin

**Comment by vasily-kirichenko** [comment]
Done.

**Comment by KevinRansom** [comment]
Thanks for taking care of this

Kevin

**Pull request merged**